plugins/verifier/sbom/sbom.go

```diff
@@ -95,29 +96,84 @@ func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, refe
 			}, err
 		}
 
-		switch mediaType {
+		switch artifactType {
 		case SpdxJSONMediaType:
-			return processSpdxJSONMediaType(input.Name, refBlob)
+			return processSpdxJSONMediaType(input.Name, refBlob, input.DisallowedLicenses, input.DisallowedPackages)
 		default:
 		}
 	}
 
 	return &verifier.VerifierResult{
 		Name:      input.Name,
 		IsSuccess: false,
-		Message:   fmt.Sprintf("Unsupported mediaType: %s", mediaType),
+		Message:   fmt.Sprintf("Unsupported artifactType: %s", artifactType),
 	}, nil
 }
 
-func processSpdxJSONMediaType(name string, refBlob []byte) (*verifier.VerifierResult, error) {
+// getViolations returns the package and license violations based on the deny list
+func getViolations(spdxDoc *spdx.Document, disallowedLicenses []string, disallowedPackages []utils.PackageInfo) ([]utils.PackageLicense, []utils.PackageLicense, error) {
+	packageLicenses := utils.GetPackageLicenses(*spdxDoc)
+	// load disallowed packageInfo into a map for easier existence check
+	packageMap, packageNameMap := loadDisallowedPackagesMap(disallowedPackages)
+
+	// detect violation
+	licenseViolation, packageViolation := filterDisallowedPackages(packageLicenses, disallowedLicenses, packageMap, packageNameMap)
+	return packageViolation, licenseViolation, nil
+}
+
+// load disallowed packageInfo, and disallowed packageName into a map for easier existence check
+func loadDisallowedPackagesMap(packages []utils.PackageInfo) (map[utils.PackageInfo]struct{}, map[string]struct{}) {
+	packagesInfo := map[utils.PackageInfo]struct{}{}
+	packagesName := map[string]struct{}{}
+
+	for _, item := range packages {
+		// if the deny list item has no specific version, add to separate map
+		if len(item.Version) == 0 {
+			packagesName[item.Name] = struct{}{}
+		}
+		packagesInfo[item] = struct{}{}
+	}
+	return packagesInfo, packagesName
+}
+
+// parse through the spdx blob and returns the verifier result
+func processSpdxJSONMediaType(name string, refBlob []byte, disallowedLicenses []string, disallowedPackages []utils.PackageInfo) (*verifier.VerifierResult, error) {
 	var err error
-	var doc *v2_3.Document
-	if doc, err = jsonLoader.Read(bytes.NewReader(refBlob)); doc != nil {
+	var spdxDoc *v2_3.Document
+	if spdxDoc, err = jsonLoader.Read(bytes.NewReader(refBlob)); spdxDoc != nil {
+		if len(disallowedLicenses) != 0 || len(disallowedPackages) != 0 {
+			packageViolation, licenseViolation, err := getViolations(spdxDoc, disallowedLicenses, disallowedPackages)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get SBOM violation %w", err)
+			}
+
+			var extensionData = make(map[string]interface{})
+			extensionData[CreationInfo] = spdxDoc.CreationInfo
+			if licenseViolation != nil {
```

Comment:
do we need to check for zero length here too?

Reply:
fixed. thanks!